client/app/visualizations/map/MapEditor.jsx

```diff
@@ -0,0 +1,212 @@
+          <div className="form-group">
+            <label className="control-label">Latitude Column Name</label>
+            <Select
+              name="form-control"
+              required
+              clearable={false}
+              value={opts.latColName}
+              placeholder="Choose column..."
+              options={columnOptions([opts.classify, opts.lonColName])}
+              onChange={this.updateLatColName}
+            />
+          </div>
+
+          <div className="form-group">
+            <label className="control-label">Longitude Column Name</label>
+            <Select
+              name="form-control"
+              required
+              clearable={false}
+              value={opts.lonColName}
+              placeholder="Choose column..."
+              options={columnOptions([opts.classify, opts.latColName])}
+              onChange={this.updateLonColName}
+            />
+          </div>
+
+          <div className="form-group">
+            <label className="control-label">Group By</label>
+            <Select
+              name="form-control"
+              required
+              clearable={false}
+              value={opts.classify}
+              placeholder="Choose column..."
+              options={concat(columnOptions([opts.lonColName, opts.latColName]), [{ value: 'none', label: 'none' }])}
+              onChange={this.updateClassify}
+            />
+          </div>
+        </div>
+      ),
+      groups: (
+        <div className="m-b-10">
+          <table className="table table-condensed col-table">
+            <thead>
+              <th>Name</th>
+              <th>Color</th>
+            </thead>
+            <tbody>
+              {map(groups, (options, name) => (
+                <tr key={name}>
+                  <td>{name}</td>
+                  <td>
+                    <input
+                      className="form-control"
+                      type="color"
+                      value={options.color}
+                      onChange={e => this.updateColor(name, e.target.value)}
+                    />
+                  </td>
+                </tr>))}
+            </tbody>
+          </table>
+        </div>
+
+      ),
+      map: (
+        <div className="m-t-10 m-b-10">
+          <div className="checkbox">
+            <label>
+              <input type="checkbox" checked={opts.clusterMarkers} onChange={this.updateClusterMarkers} />
+              <i className="input-helper" /> Cluster Markers
+            </label>
+          </div>
+
+          <div className="form-group">
+            <label className="control-label">Map Tiles</label>
+            <select value={opts.mapTileUrl} className="form-control" onChange={this.updateMapTileUrl}>
+              {mapTiles.map(tile => <option key={tile.url} value={tile.url}>{tile.name}</option>)}
+            </select>
+          </div>
+        </div>
+      ),
+    };
+    return (
+      <div>
+        <ul className="tab-nav">
```

Comment:
Instead of implementing the tab navigation logic everywhere (like we did in Angular), we can use [Ant's Tabs component](https://ant.design/components/tabs/). Also we should consider splitting each tab into each own component.